app/src/main/java/com/appdev/eateryblueandroid/ui/components/home/PaymentMethodFilter.kt

```diff
@@ -0,0 +1,219 @@
+    saveFilter: (filter: List<String>) -> Unit,
+    hide: () -> Unit
+) {
+    Column(
+        modifier = Modifier.padding(start = 16.dp, end = 16.dp, top = 24.dp, bottom = 24.dp)
+    ) {
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+            Text(
+                text = "Payment Methods",
+                textStyle = TextStyle.HEADER_H3,
+                modifier = Modifier.padding(bottom = 12.dp)
+            )
+            CircularBackgroundIcon(
+                icon = painterResource(id = R.drawable.ic_x),
+                clickable = true,
+                onTap = {
+                    saveFilter(selectedFilter.value.filter { it != "Payment Options" })
+                    hide()
+                },
+                iconWidth = 12.dp,
+                iconHeight = 12.dp,
+                backgroundSize = 40.dp
+            )
+        }
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(vertical = 8.dp, horizontal = 16.dp),
+            horizontalArrangement = Arrangement.SpaceEvenly,
+        ) {
+            Column(
+                modifier = Modifier.width(90.dp),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                CircularBackgroundIcon(
+                    icon = painterResource(id = R.drawable.ic_payment_swipes),
+                    iconTint = colorResource(id = if (selectedFilter.value.contains("Meal swipes")) R.color.white else R.color.gray05),
+                    backgroundTint = colorResource(id = if (selectedFilter.value.contains("Meal swipes")) R.color.eateryBlue else R.color.gray00),
+                    onTap = { toggleFilter("Meal swipes") },
+                    clickable = true,
+                    iconWidth = 36.dp,
+                    iconHeight = 36.dp,
+                    backgroundSize = 64.dp
+                )
+                Text(
+                    text = "Meal Swipes",
+                    textStyle = TextStyle.LABEL_SEMIBOLD,
+                    modifier = Modifier.padding(top = 8.dp, bottom = 8.dp)
+                )
+            }
+            Column(
+                modifier = Modifier.width(90.dp),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                CircularBackgroundIcon(
+                    icon = painterResource(id = R.drawable.ic_payment_brbs),
+                    iconTint = colorResource(id = if (selectedFilter.value.contains("BRBs")) R.color.white else R.color.gray05),
+                    backgroundTint = colorResource(id = if (selectedFilter.value.contains("BRBs")) R.color.red else R.color.gray00),
+                    onTap = { toggleFilter("BRBs") },
+                    clickable = true,
+                    iconWidth = 36.dp,
+                    iconHeight = 36.dp,
+                    backgroundSize = 64.dp
```

Comment:
Duplicate code as above on lines 64-72 -- not sure if you could set defaults or something?